---
title: "Genomic Profiling of Cancer Subtypes"
author: "Your Name"
date: "`r format(Sys.time(), '%Y-%m-%d')`"
output:
  html_document:
    toc: true
    toc_depth: 3
---


## Overview

This R Markdown document provides an analysis of genomic profiling of cancer subtypes using the Golub dataset. The project includes clustering analysis, correlation analysis, differential expression analysis, and visualization of results.

```{r setup, include=FALSE}
# Load required packages
library(multtest)
library(dplyr)
library(cluster)
library(corrplot)

# Load the Golub dataset
data(golub, package = "multtest")
golub_gnames <- golub.gnames[, 2]
```


## Data Loading and Preprocessing
```{r}
# Display summary of the dataset
summary(golub)

# Display structure of the dataset
str(golub)

# Display the first few rows of the dataset
head(golub)

# Display the last few rows of the dataset
tail(golub)
```


## Clustering Analysis
Hierarchical Clustering
```{r}
# Select specific genes for clustering analysis
CCND3 <- grep("CCND3 Cyclin D3", golub_gnames)
CCND3_data <- golub[CCND3, ]

# Perform hierarchical clustering
hc_single <- hclust(dist(CCND3_data))
hc_ward <- hclust(dist(CCND3_data), method = "ward.D2")

# Plot dendrograms
plot(hc_single, main = "Single Linkage Hierarchical Clustering", xlab = "Patients", ylab = "Distance")
plot(hc_ward, main = "Ward Linkage Hierarchical Clustering", xlab = "Patients", ylab = "Distance")

```

## K-means Clustering
```{r}
# Perform k-means clustering
k <- 2  # Number of clusters
kmeans_clusters <- kmeans(CCND3_data, centers = k)

# Visualize k-means clusters
clusplot(CCND3_data, kmeans_clusters$cluster, color = TRUE, shade = TRUE, labels = 2, lines = 0)
```

## Correlation Analysis
```{r}
# Correlation matrix of all genes in the dataset
cor_matrix <- cor(golub)

# Plot correlation matrix using corrplot
corrplot(cor_matrix, method = "color", type = "upper", order = "hclust", addrect = 2)
```


## Differential Expression Analysis
```{r}
# Perform differential expression analysis
group <- golub.cl
gene_expression <- golub
de_results <- apply(gene_expression, 1, function(x) t.test(x ~ group)$p.value)

# Adjust p-values for multiple testing using FDR
de_results_adj <- p.adjust(de_results, method = "fdr")

# Identify significant genes
significant_genes <- which(de_results_adj < 0.05)
significant_genes_names <- golub_gnames[significant_genes]

# Output significant genes
cat("Significant genes:", significant_genes_names, "\n")

```

